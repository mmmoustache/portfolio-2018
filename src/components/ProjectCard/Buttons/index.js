import styled from "styled-components";
import { Breakpoints } from '../../../utils/breakpoints';
import { convertRemToPixels as rem } from '../../../utils/utils';

export default styled.div`
  display: flex;
  margin-top: ${rem('30px')};
  flex-flow: wrap;
  
  & > *:last-child {
    ${
      props => props.multiple && `
        margin-top: ${rem('20px')};
      `
    }
  }

  @media ${Breakpoints.small} {
    flex-flow: initial;

    & > *:last-child {
      ${
        props => props.multiple && `
          margin-left: ${rem('20px')};
          margin-top: 0;
        `
      }
    }
  }
`;
