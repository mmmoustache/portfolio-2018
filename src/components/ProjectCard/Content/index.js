import styled from "styled-components";
import { Breakpoints } from '../../../utils/breakpoints';
import { Colours } from '../../../utils/colours';
import { Timings, Durations } from '../../../utils/transitions';
import { convertRemToPixels as rem } from '../../../utils/utils';

const ContentOffset = rem('40px');

export const Content = styled.div`
  display: inline-block;
  opacity: 0;
  position: relative;
  transform: translateY(${ContentOffset});
  transition:  opacity ${Durations.default} ${Timings.default}, transform ${Durations.default} ${Timings.default};
  transition-delay: ${Durations.fast};
  width: 100%;

  @media ${Breakpoints.large} {
    width: 60%;
  }

  ${
    props => props.isVisible && `
      opacity: 1;
      transform: translateY(0);
    `
  }
`;

export const ContentInner = styled.div`
  color: ${Colours.white};
`;
