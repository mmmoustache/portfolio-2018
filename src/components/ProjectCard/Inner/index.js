import styled from "styled-components";
import { Breakpoints } from '../../../utils/breakpoints';
import { Layouts } from '../../../utils/layouts';
import { convertRemToPixels as rem } from '../../../utils/utils';

export default styled.div`
  margin: 0 auto;
  max-width: ${Layouts.constraint};
  padding: ${rem('20px')};

  @media ${Breakpoints.medium} {
    position: relative;
    top: 50%;
    transform: translateY(-50%);
  }
`;
