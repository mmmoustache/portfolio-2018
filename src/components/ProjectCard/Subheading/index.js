import { convertRemToPixels as rem } from '../../../utils/utils';
import { Breakpoints } from '../../../utils/breakpoints';
import H3 from "../../H3";

export default H3.extend`
  margin: 0;
  
  @media ${Breakpoints.medium} {
    margin-bottom: ${rem('20px')};
  }
`;
