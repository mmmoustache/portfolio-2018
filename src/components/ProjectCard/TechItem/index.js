import styled from "styled-components";
import { convertRemToPixels as rem } from '../../../utils/utils';

export default styled.li`
  display: inline-block;
  margin-right: ${rem('15px')};
`;
