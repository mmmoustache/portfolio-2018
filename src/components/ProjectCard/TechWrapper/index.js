import styled from "styled-components";
import { convertRemToPixels as rem } from '../../../utils/utils';
import { Breakpoints } from '../../../utils/breakpoints';

export default styled.ul`
  margin-bottom: ${rem('20px')};

  @media ${Breakpoints.medium} {
    margin-bottom: 0;
  }
`;
