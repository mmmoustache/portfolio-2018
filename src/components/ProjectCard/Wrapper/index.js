import styled from "styled-components";
import { Colours } from '../../../utils/colours';
import { Timings, Durations } from '../../../utils/transitions';
import { Breakpoints } from '../../../utils/breakpoints';
import { convertRemToPixels as rem } from '../../../utils/utils';

export default styled.div`
  background: ${Colours.black};
  box-sizing: border-box;
  overflow: hidden;
  padding: ${rem('60px')} 0;
  position: relative;
  transition-delay: 1s;
  transition: background ${Durations.default} ${Timings.default};

  @media ${Breakpoints.medium} {
    height: 100vh;
  }

  ${
    props => props.isVisible && `
      background: ${Colours.secondary};
    `
  }

  ${
    props => props.isVisible && props.isOdd && `
      background: ${Colours.primary};
    `
  }
`;
