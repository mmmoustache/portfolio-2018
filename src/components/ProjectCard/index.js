import React from 'react';
import styled from "styled-components";
import Button from "../Button";
import H2 from "../H2";
import H3 from "../H3";
import Text from "../Text";

const Wrapper = styled.div`
  background-position: center;
  background-size: cover;

  /* background-image: ${ props => props.image && `url(${props.image})` };*/
`;

const Inner = styled.div`
  display: flex;
  margin: 0 auto;
  max-width: 1200px;
  padding: 20px;
  position: relative;
  top: 50%;
  transform: translateY(-50%); 
`;

const TechList = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`;

const Figure = styled.div`
  width: 40%;
`;

const TechItem = styled.li`
  display: inline-block;
`;

const Content = styled.div`
  background: #fff;
  display: block;
  position: relative;
  width: 60%;

  &::after {
    border: 10px solid red;
    content: '';
    height: 100%;
    width: 60%;
    top: -50px;
    right: -30px;
    position: absolute;
    z-index: -1;
  }
`;

const ProjectCard = ({ title, description, image_url, technologies, url, repo }) => 
  <Wrapper image={image_url}>
    {
      technologies &&
        <TechList>
          { technologies.map(e => <TechItem key={e}><H3>{e}</H3></TechItem>) }
        </TechList>
    }

    <H2>{title}</H2>
    <Text>
      <p>{description}</p>
    </Text>
    { url && <Button href={url}>View Site</Button> }
    { repo && <Button href={repo}>Code</Button> }
  </Wrapper>
;

export default ProjectCard;
