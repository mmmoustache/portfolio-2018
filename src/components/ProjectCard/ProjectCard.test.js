import React from 'react';
import renderer from 'react-test-renderer';
import ProjectCard from "../ProjectCard";

describe('Component: ProjectCard', () => {

  it('Should render successfully', () => {
    const component = renderer.create(
      <ProjectCard 
        title="Test" 
        description="Test"
        image_url="Test"
        technologies={[]} 
        url="Test" 
        repo="Test" 
        isVisible
        isOdd
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
