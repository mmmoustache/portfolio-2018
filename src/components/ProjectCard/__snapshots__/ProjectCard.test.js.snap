// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProjectCard Should render successfully 1`] = `
<div
  className="sc-EHOje idMEnW"
>
  <div
    className="sc-jTzLTM blZols"
  >
    <div
      className="sc-jzJRlG iqgKRc"
    />
    <div
      className="sc-cSHVUG fRZEWR"
    />
    <div
      className="sc-kAzzGY bAiDWj"
    />
    <div
      className="sc-chPdSV cAeglF"
    />
  </div>
  <div
    className="sc-ifAKCX kuTWZI"
  >
    <div
      className="sc-gqjmRU cZBFKU"
    >
      <div
        className="sc-VigVT bjRZMr"
      >
        <ul
          className="sc-htoDjs iOfqWF"
        />
        <h2
          className="sc-gZMcBi FOban"
        >
          Test
        </h2>
        <div
          className="sc-bxivhb erkEfj"
        >
          <p>
            Test
          </p>
        </div>
      </div>
      <div
        className="sc-bZQynM fDfFdw"
      >
        <a
          className="sc-htpNat deSbaL"
          href="Test"
        >
          <span
            className="sc-bwzfXH cbEDiN"
          >
            <h3
              className="sc-bdVaJa bjYFto"
            >
              View Site
            </h3>
          </span>
        </a>
        <a
          className="sc-htpNat deSbaL"
          href="Test"
        >
          <span
            className="sc-bwzfXH cbEDiN"
          >
            <h3
              className="sc-bdVaJa bjYFto"
            >
              View Code
            </h3>
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
